server/src/main/java/org/elasticsearch/search/dfs/ProfilingKnnByteVectorQuery.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.dfs;
+
+import org.apache.lucene.search.KnnByteVectorQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+
+public class ProfilingKnnByteVectorQuery extends KnnByteVectorQuery {
```

Comment:
All these `Profiling*` things should also implement an interface. I am thinking something like

`implements ProfiledQuery`, then the interface has something like:

```
void profile(QueryProfiler queryProfiler);
```
And it can update the profiler with any information it collected.

Reply:
Updated